controllers/home.js

```diff
@@ -1,10 +1,30 @@
 "use strict";
 
+const crypto = require("crypto");
 const { URL } = require("url");
+const uuidv4 = require("uuid/v4");
 
+const AppConstants = require("../app-constants");
 const HIBP = require("../hibp");
 const sha1 = require("../sha1-utils");
 
+
+function _generatePageToken(req) {
+  const pageToken = {ip: req.ip, date: new Date(), nonce: uuidv4()};
```

Comment:
just to sanity check: `req.ip` contains the client ip from the x-forwarded-for, not the load balancer?

Reply:
Correct, because we've got [`app.enable("trust proxy");`](https://expressjs.com/en/4x/api.html#trust.proxy.options.table)

Reply:
From that doc: `If true, the client’s IP address is understood as the left-most entry in the X-Forwarded-* header.`

Which is a big fail, because we know bots inject x-forwarded-for headers to pass random ips as the leftmost entry. @Micheletto can confirm that our glb doesn't sanitize those, which is sad, but effectively means this control won't work.

Reply:
Ok, that library supports `Trusting the nth hop from the front-facing proxy server as the client.` Is that how we're currently setup?

Reply:
Doesn't look like it? I think this should be a number instead of `true` now: https://github.com/mozilla/blurts-server/blob/1e2bc90b3e20a0a59438c0f6b9bda9241b831ed8/server.js#L136-L140

Sanitizing XFF seems like pretty basic functionality, disappointing that GLB doesn't do that.

Reply:
> @Micheletto can confirm that our glb doesn't sanitize those, which is sad, but effectively means this control won't work.

That is correct.  We could set a header that is passed to the app with the real-ip though.

Reply:
Nope, but I can set it to whatever N we want. Do we know it will always be the *n*th hop from the front-facing proxy server?

Reply:
I'm going to drop the `trust proxy` setting here, and we're going to have nginx explicitly set a custom header `X-Real-IP` that I will read from.